socketdock/backend.py

```diff
@@ -7,6 +7,13 @@
 class Backend(ABC):
     """Backend interface for SocketDock."""
 
+    @abstractmethod
+    async def inbound_connected(
```

Comment:
For consistency, I would name this `socket_connected` so it matches the existing `socket_disconnected`.